tests/subsys/audio_module/src/bad_param_test.c

```diff
@@ -0,0 +1,1171 @@
+		      -ECANCELED, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_disconnect_bad_state)
+{
+	int ret;
+
+	test_initialise_description(&test_description_tx, "Module Test 1", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+	test_initialise_description(&test_description_rx, "Module Test 2", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+
+	test_initialise_handle(&handle_tx, "TEST Disconnect 1", &test_description_tx,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+	test_initialise_handle(&handle_rx, "TEST Disconnect 2", &test_description_rx,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+
+	ret = audio_module_disconnect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Disconnect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	handle_tx.state = AUDIO_MODULE_STATE_CONFIGURED;
+	handle_rx.state = AUDIO_MODULE_STATE_UNDEFINED;
+	ret = audio_module_disconnect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Disconnect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	handle_tx.state = AUDIO_MODULE_STATE_UNDEFINED;
+	handle_rx.state = AUDIO_MODULE_STATE_CONFIGURED;
+	ret = audio_module_disconnect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Disconnect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_disconnect_null)
+{
+	int ret;
+
+	test_initialise_description(&test_description_tx, "Module Test 1", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+	test_initialise_description(&test_description_rx, "Module Test 2", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+
+	test_initialise_handle(&handle_tx, "TEST Disconnect 1", &test_description_tx,
+			       AUDIO_MODULE_STATE_CONFIGURED, NULL, NULL);
+	test_initialise_handle(&handle_rx, "TEST Disconnect 2", &test_description_rx,
+			       AUDIO_MODULE_STATE_CONFIGURED, NULL, NULL);
+
+	ret = audio_module_disconnect(NULL, NULL, false);
+	zassert_equal(ret, -EINVAL, "Disconnect set function did not return -EINVAL (%d): ret %d",
+		      -EINVAL, ret);
+
+	ret = audio_module_disconnect(NULL, &handle_rx, false);
+	zassert_equal(ret, -EINVAL, "Disconnect set function did not return -EINVAL (%d): ret %d",
+		      -EINVAL, ret);
+
+	ret = audio_module_disconnect(&handle_tx, NULL, false);
+	zassert_equal(ret, -EINVAL, "Disconnect set function did not return -EINVAL (%d): ret %d",
+		      -EINVAL, ret);
+
+	ret = audio_module_disconnect(&handle_tx, &handle_tx, false);
+	zassert_equal(ret, -EINVAL, "Disconnect set function did not return -EINVAL (%d): ret %d",
+		      -EINVAL, ret);
+
+	ret = audio_module_disconnect(&handle_tx, &handle_rx, true);
+	zassert_equal(ret, -EINVAL, "Disconnect set function did not return -EINVAL (%d): ret %d",
+		      -EINVAL, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_connect_bad_type)
+{
+	int ret;
+
+	test_initialise_description(&test_description_tx, "Module Test 1",
```

Comment:
Would it be more natural to call the modules` Module test tx/rx` ?

Reply:
I have reused the tx and rx description variables. The description variable names really should really change, will correct for all tests.